### PR TITLE
Scrollable - Add top level css classes

### PR DESCRIPTION
**Pull request by novsstation**
Add top level css classes for scrollview markup

### CODE REVIEW

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx [hunk 1]
```diff
@@ -1040,10 +1046,16 @@ jest.mock('../../../../core/utils/scroll_rtl_behavior', () => () => testBehavior
       describe('Getters', () => {
         describe('cssClasses', () => {
           it('should add scrolling classes by default', () => {
-            const { cssClasses } = new Scrollable({});
-            expect(cssClasses).toEqual(expect.stringMatching('dx-scrollable'));
-            expect(cssClasses).toEqual(expect.stringMatching('dx-scrollable-native'));
-            expect(cssClasses).toEqual(expect.stringMatching('dx-scrollable-native-generic'));
+            const instance = new Scrollable({});
+            expect(instance.cssClasses).toEqual(expect.stringMatching('dx-scrollable'));
+            expect(instance.cssClasses).toEqual(expect.stringMatching('dx-scrollview'));
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: 'dx-scrollview-content' is added only for ScrollView component in old code, it should not be added for any scrollable

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx [hunk 2]
```diff
@@ -1040,10 +1046,16 @@ jest.mock('../../../../core/utils/scroll_rtl_behavior', () => () => testBehavior
...
+
+            if (instance instanceof ScrollableNative) {
+              expect(instance.cssClasses).toEqual(expect.stringMatching('dx-scrollable-native'));
+              expect(instance.cssClasses).toEqual(expect.stringMatching(`dx-scrollable-native-${devices.current().platform}`));
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: imho we should force devices.current().platform to return some value and check that there is a class with that value
const oldDevice = devices._currentDevice;
devices.device = { platform: 'platform1'};

expect(instance.cssClasses).toEqual(expect.stringMatching(`dx-scrollable-native-patform1`));

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx [hunk 3]
```diff
@@ -42,10 +43,10 @@ jest.mock('../../../../core/utils/scroll_rtl_behavior', () => () => testBehavior
 }].forEach(({ viewFunction, Scrollable }) => {
   describe('Scrollable', () => {
     describe('Render', () => {
-      it('should render scrollable content', () => {
+      it('should render scrollView content', () => {
         const scrollable = shallow(viewFunction({ props: { } } as any) as JSX.Element);
-        const scrollableContent = scrollable.find('.dx-scrollable-wrapper > .dx-scrollable-container > .dx-scrollable-content');
-        expect(scrollableContent.exists()).toBe(true);
+        const scrollViewContent = scrollable.find('.dx-scrollable-wrapper > .dx-scrollable-container > .dx-scrollable-content > .dx-scrollview-content');
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: 'dx-scrollview-content' is added only for ScrollView component, it should not be added for any scrollable:
```
const ScrollView = Scrollable.inherit(isServerSide ? scrollViewServerConfig : {
    _initContent: function() {
        const $content = $('<div>').addClass(SCROLLVIEW_CONTENT_CLASS);
        this._$content.wrapInner($content);
    },
```

New prop 'isScrollView' can help.

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx [hunk 4]
```diff
@@ -67,6 +68,11 @@ jest.mock('../../../../core/utils/scroll_rtl_behavior', () => () => testBehavior
         expect(bottomPocket.exists()).toBe(true);
       });
 
+      it('Scrollable should have dx-scrollable-disabled if disabled', () => {
+        const instance = new Scrollable({ disabled: true });
+        expect(instance.cssClasses).toEqual(expect.stringMatching('dx-scrollable-disabled'));
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: add this config:
```
        const instance = new Scrollable({ disabled: false });
        expect(instance.cssClasses).toEqual(expect.not.stringMatching('dx-scrollable-disabled'));
```

File: js/renovation/ui/scroll_view/scrollable_native.tsx [hunk 5]
```diff
@@ -50,7 +52,9 @@ export const viewFunction = ({
       <div className={SCROLLABLE_CONTAINER_CLASS} ref={containerRef as any}>
         <div className={SCROLLABLE_CONTENT_CLASS} ref={contentRef as any}>
           {forceGeneratePockets && <div className={SCROLLVIEW_TOP_POCKET_CLASS} />}
-          {children}
+          <div className={SCROLLVIEW_CONTENT_CLASS}>
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/scrollable_native.tsx]: 'dx-scrollview-content' and wrapping 'div' are added only for ScrollView component in old code, it should not be added for any scrollable

File: js/renovation/ui/scroll_view/scrollable_simulated.tsx [hunk 6]
```diff
@@ -50,7 +51,9 @@ export const viewFunction = ({
       <div className={SCROLLABLE_CONTAINER_CLASS} ref={containerRef as any}>
         <div className={SCROLLABLE_CONTENT_CLASS} ref={contentRef as any}>
           {forceGeneratePockets && <div className={SCROLLVIEW_TOP_POCKET_CLASS} />}
-          {children}
+          <div className={SCROLLVIEW_CONTENT_CLASS}>
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/scrollable_simulated.tsx]: Again wrapping div from ScrollView

File: js/renovation/ui/scroll_view/scrollable_simulated.tsx [hunk 7]
```diff
@@ -171,11 +174,12 @@ export class ScrollableSimulated extends JSXComponent<ScrollableInternalPropsTyp
   }
 
   get cssClasses(): string {
-    const { direction, classes } = this.props;
+    const { direction, classes, disabled } = this.props;
 
     const classesMap = {
-      'dx-scrollable dx-scrollable-native dx-scrollable-renovated dx-scrollable-native-generic': true,
+      'dx-scrollable dx-scrollview dx-scrollable-simulated dx-scrollable-renovated': true,
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/scrollable_simulated.tsx]: sx-scrollview should not be added always, it is for ScrollView component

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx (hunk 3 above)
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: 'should add scrolling classes by default' test should check 'dx-scrollable-wrapper'

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx [hunk 8]
```diff
@@ -67,6 +68,11 @@ jest.mock('../../../../core/utils/scroll_rtl_behavior', () => () => testBehavior
         expect(bottomPocket.exists()).toBe(true);
       });
 
+      it('Scrollable should have dx-scrollable-disabled if disabled', () => {
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: "expect(instance.cssClasses).toEqual(expect.stringMatching('dx-scrollable-disabled'));" check should be in the describe('cssClasses' module

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx (hunk 3 above)
Comment (novsstation) [reply js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: Yep, you are absolutely right.
Fixed https://github.com/DevExpress/DevExtreme/pull/15636/commits/3371f7a4122918b13a658b8a49751be74f06fa71

Comment (novsstation) [reply js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: Yep, great idea.
Fixed https://github.com/DevExpress/DevExtreme/pull/15636/commits/3371f7a4122918b13a658b8a49751be74f06fa71

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx (hunk 8 above)
Comment (novsstation) [reply js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: Yep, fixed https://github.com/DevExpress/DevExtreme/pull/15636/commits/3371f7a4122918b13a658b8a49751be74f06fa71

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx (hunk 4 above)
Comment (novsstation) [reply js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: Fixed https://github.com/DevExpress/DevExtreme/pull/15636/commits/3371f7a4122918b13a658b8a49751be74f06fa71

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx (hunk 1 above)
Comment (novsstation) [reply js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: Yep fixed.
Now this class will be added only for ScrollView component https://github.com/DevExpress/DevExtreme/pull/15636/commits/3371f7a4122918b13a658b8a49751be74f06fa71

File: js/renovation/ui/scroll_view/scrollable_native.tsx (hunk 5 above)
Comment (novsstation) [reply js/renovation/ui/scroll_view/scrollable_native.tsx]: Yep, now this wrapper will be added only inside Scrollview component https://github.com/DevExpress/DevExtreme/pull/15636/commits/3371f7a4122918b13a658b8a49751be74f06fa71

File: js/renovation/ui/scroll_view/scrollable_simulated.tsx (hunk 6 above)
Comment (novsstation) [reply js/renovation/ui/scroll_view/scrollable_simulated.tsx]: Yep fixed https://github.com/DevExpress/DevExtreme/pull/15636/commits/3371f7a4122918b13a658b8a49751be74f06fa71

File: js/renovation/ui/scroll_view/scrollable_simulated.tsx (hunk 7 above)
Comment (novsstation) [reply js/renovation/ui/scroll_view/scrollable_simulated.tsx]: Fixed https://github.com/DevExpress/DevExtreme/pull/15636/commits/3371f7a4122918b13a658b8a49751be74f06fa71

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx (hunk 2 above)
Comment (novsstation) [reply js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: Yep, fixed with using of mocks 
https://github.com/DevExpress/DevExtreme/pull/15636/commits/8d23a20941faa8c5398c7704b8b5b2c470f610f3

File: js/renovation/ui/scroll_view/scrollable_utils.ts [hunk 9]
```diff
@@ -12,8 +12,10 @@ export const DIRECTION_BOTH = 'both';
 export const SCROLLABLE_CONTENT_CLASS = 'dx-scrollable-content';
 export const SCROLLABLE_WRAPPER_CLASS = 'dx-scrollable-wrapper';
 export const SCROLLABLE_CONTAINER_CLASS = 'dx-scrollable-container';
+export const SCROLLVIEW_CONTENT_CLASS = 'dx-scrollview-content';
 export const SCROLLVIEW_TOP_POCKET_CLASS = 'dx-scrollview-top-pocket';
 export const SCROLLVIEW_BOTTOM_POCKET_CLASS = 'dx-scrollview-bottom-pocket';
+export const SCROLLABLE_DISABLED_CLASS = 'dx-scrollable-wrapper';
```
Comment (EugeniyKiyashko) [inline js/renovation/ui/scroll_view/scrollable_utils.ts]: Seems we need change className to dx-scrollable-disabled

Comment (novsstation) [reply js/renovation/ui/scroll_view/scrollable_utils.ts]: Fixed

**Review by EugeniyKiyashko — DISMISSED** [review]

**Review by KuznetsovVN — DISMISSED** [review]

File: js/renovation/ui/scroll_view/scrollable_native.tsx [hunk 10]
```diff
@@ -50,7 +54,8 @@ export const viewFunction = ({
       <div className={SCROLLABLE_CONTAINER_CLASS} ref={containerRef as any}>
         <div className={SCROLLABLE_CONTENT_CLASS} ref={contentRef as any}>
           {forceGeneratePockets && <div className={SCROLLVIEW_TOP_POCKET_CLASS} />}
-          {children}
+          {needScrollViewWrappers && <div className={SCROLLVIEW_CONTENT_CLASS}>{children}</div>}
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/scrollable_native.tsx]: 'needScrollViewWrappers' doesn't look good for me
Let's rename similar to 'SCROLLVIEW_CONTENT': 'needScrollViewContentWrapper'

File: js/renovation/ui/scroll_view/scrollable_native.tsx [hunk 11]
```diff
@@ -171,11 +176,12 @@ export class ScrollableNative extends JSXComponent<ScrollableInternalPropsType>(
   }
 
   get cssClasses(): string {
-    const { direction, classes } = this.props;
+    const { direction, classes, disabled } = this.props;
 
     const classesMap = {
-      'dx-scrollable dx-scrollable-native dx-scrollable-renovated dx-scrollable-native-generic': true,
+      [`dx-scrollable dx-scrollable-native dx-scrollable-native-${devices.current().platform} dx-scrollable-renovated`]: true,
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/scrollable_native.tsx]: Why 'devices.current().platform' while it was 'devices.real().platform'?

```
        const device = **devices.real();**
        const deviceType = device.platform;

        this._$element
            .addClass(SCROLLABLE_NATIVE_CLASS)
            .addClass(SCROLLABLE_NATIVE_CLASS + '-' + deviceType)
```

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: js/renovation/ui/scroll_view/scroll_view.tsx [hunk 12]
```diff
@@ -38,6 +38,7 @@ export const viewFunction = (viewModel: ScrollView): JSX.Element => {
       {...props}
       // eslint-disable-next-line react/jsx-props-no-spreading
       {...restAttributes}
+      needScrollViewWrappers
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/scroll_view.tsx]: Where it is initialized to 'true' to work like old code?
```
    _initContent: function() {
        const $content = $('<div>').addClass(SCROLLVIEW_CONTENT_CLASS);
        this._$content.wrapInner($content);
    },
```

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx [hunk 13]
```diff
@@ -1039,11 +1057,19 @@ jest.mock('../../../../core/utils/scroll_rtl_behavior', () => () => testBehavior
     describe('Logic', () => {
       describe('Getters', () => {
         describe('cssClasses', () => {
-          it('should add scrolling classes by default', () => {
-            const { cssClasses } = new Scrollable({});
-            expect(cssClasses).toEqual(expect.stringMatching('dx-scrollable'));
-            expect(cssClasses).toEqual(expect.stringMatching('dx-scrollable-native'));
-            expect(cssClasses).toEqual(expect.stringMatching('dx-scrollable-native-generic'));
+          ['desktop', 'ios', 'android'].forEach((platform) => {
+            it('should add scrolling classes by default', () => {
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: Test name should contain platform name

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx [hunk 14]
```diff
@@ -1039,11 +1057,19 @@ jest.mock('../../../../core/utils/scroll_rtl_behavior', () => () => testBehavior
...
+              (devices.current as Mock).mockImplementation(() => ({ platform }));
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: Are you sure we need Mock here?
Does 'devices.current = () => ({ platform })' meet our needs?

File: js/renovation/ui/scroll_view/scrollable_native.tsx (hunk 11 above)
Comment (novsstation) [reply js/renovation/ui/scroll_view/scrollable_native.tsx]: Yep, my mistake. Fixed https://github.com/DevExpress/DevExtreme/pull/15636/commits/9d8f0a12a9faf520c78e6ccdffcab5523984d43c

File: js/renovation/ui/scroll_view/scroll_view.tsx (hunk 12 above)
Comment (novsstation) [reply js/renovation/ui/scroll_view/scroll_view.tsx]: These changes are consistent. It's value already is true

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx (hunk 14 above)
Comment (novsstation) [reply js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: Fixed

File: js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx (hunk 13 above)
Comment (novsstation) [reply js/renovation/ui/scroll_view/__tests__/scrollable.test.tsx]: Yep, fixed

**Review by IgnatovDan — APPROVED** [review]

**Review by EugeniyKiyashko — APPROVED** [review]

**Pull request merged**